awkward-pybind11/awkward/pybind11/__init__.py

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env python
+
+# Copyright (c) 2019, IRIS-HEP
```

Comment:
IRIS-HEP is not a legal entity, so it can’t hold a copyright.  Check with your employer/institution about their local policies.

Reply:
This is a long-standing thing I've been wondering if I've been doing it right. The license (BSD-3) says that anyone can use the software, but we still need to put something in the "who wrote it" slot. I'm not the only author (as you can see from this pull request) and even IRIS-HEP is inappropriate because some of the committers are not IRIS-HEP members. I doubt it's the university for the same reason.

I expect Princeton does have someone I can ask about legal questions like, "what name should go in the copyright field?" but is want to make it clear that this is software with contributions from many non-university members. Is there a word for that I should use?

Reply:
Tradition is to have the copyright statement be about the original author of the file.  So, if you wrote the file, the copyright statement should say Princeton.

At most universities, copyright for faculty research is retained by the faculty.  Works for hire - ie, when you are assigned to do something specific  as opposed to doing research - result in the copyright being owned by the university.

So, I think the best practice would be to put your name.

Note none of this is a legal analysis - and an incorrect statement in a source code file likely doesn’t make a definitive statement.  So, take a good guess between you individually and Princeton, use that, and have no worry that an incorrect guess has a significant impact.

Reply:
Goes without saying that, irrespective of the statement at the top of the file, individuals still retain their own copyright to contributions.  So, having these headers are more a historical practice.

Reply:
For these projects overall—uproot, awkward, etc.—I can legitimately put my name as copyright holder because I'm the first and main author (and not work-for-hire). However, some of the individual files, like the ones in this PR, weren't even created by or primarily edited by me, which is why I was uncomfortable with requiring my name at the top of each of them.

So I have an idea: I'll put my name in the copyright slot of the main LICENSE file at the root of the repo but remove the copyright lines from all of the source files, so that I'm not claiming every one individually. I'll leave the rest of the license—everything but the copyright—at the top of each source file, or maybe replace the whole block of text with a URL to the main LICENSE (since they exist as a reminder, in case the source file gets separated from the repo).

Reply:
FWIW, I think this kind of discussion is precisely the sort of thing it would actually be interesting to make into a topical meeting, with some presentations about how this is done in places that have thought more about it then we have and (perhaps) dredging up the previous discussions about legal entities (e.g. making the HSF into a legal entity).
IRIS-HEP is simply a funding award from NSF, and not a legal entity like you, me, Princeton or (presumably) the Morgridge Instittute. So I don't think that can hold a copyright. You could of course add entries in files to acknowledge support, e.g. some variant of the following (for Jim, for example):

Acknowledgement: support for this work was provided by NSF cooperative agreement OAC-1836650 (IRIS-HEP), grant OAC-1450377 (DIANA/HEP) and PHY-1520942 (US-CMS LHC Ops).

But this is more of courtesy than a legal statement. 

Arguably, the best we could get out of a dedicated discussion in a topical meeting would be a clear "documentation" webpage on the IRIS-HEP website that provides guidance for people for how they should/could be dealing with this. Actually having that kind of policy/guidance is something that an entity like the software institute should provide.

Reply:
That would be a good topical meeting (I don't know who we would invite to speak, though). I've been more careful with license than copyright because the license says what users are allowed to _do_, but I'd be interested in advice about copyright, too. The issue is that the code here was written by people who are not all under the same organization, but that must be a common issue for open source software.

I'm going to go ahead with my plan of setting the global (not per-file) copyright of these packages to my name and not including a per-file copyright. I will include the NSF acknowledgement with grant numbers—on each file perhaps, because the LICENSE file must be of a predetermined form and I don't want to possibly break that. Updating all the source file headers in a big sweep is not a problem because I have to do a clean-up on everything to be ready for some talks next week, anyway.

If we later, at a topical meeting, conclude that a different way to present it is better, I can change it again.

In the meantime, let's move this discussion to another channel—a mailing list or Slack—because not everyone in our group is going to see it here. This is Charlie Escott's test-problem submission for his GSoC proposal. This discussion about copyrights might be a bit confusing!

Reply:
Allow me to chip in: we kind of been through this a few times already, with no consensus. For sure our situations aren't different from other communities such as scikit-learn or Astropy - hinting at possible folks to invite.

Reply:
That's why I said it was a long-standing question. If I remember right, we didn't end up disagreeing, just didn't come to a firm conclusion. But that means that we're all in favor of uniform, professional advice.

Reply:
Yep, @jpivarski - would certainly be a great topic for a meeting raising interest from virtually everyone.